src/librawspeed/decoders/Cr2Decoder.cpp

```diff
@@ -131,11 +130,10 @@ RawImage Cr2Decoder::decodeNewFormat() {
   TiffEntry* offsets = raw->getEntry(STRIPOFFSETS);
   TiffEntry* counts = raw->getEntry(STRIPBYTECOUNTS);
 
-  LJpegPlain l(*mFile, offsets->getInt(), counts->getInt(), mRaw);
-  l.addSlices(s_width);
+  Cr2Decompressor d(*mFile, offsets->getInt(), counts->getInt(), mRaw);
```

Comment:
> Are you talking in general, or about these {jpeg,deflate} decompressors?

About all of them, actually. The difference with the LJpeg decompressors is that there you actually have a use for the internal modeling of the `AbstractLJpegDecompressor` class, because the different variants share common code. But the usage of such code could as well be modeled with a single function:
```
void decompressCr2( const Buffer& data, Buffer::size_type offset,
                      Buffer::size_type size, const RawImage& img,
                      vector<int> slicesW)
{
   Cr2Decompressor(data, offset, size, img) d;
   d.decode(slicesW);
}
```

I meant in the current code there is an object instantiation and then exactly one call to a method of that object. Whether there are several threads doing both at the same time is irrelevant to my point.